Getting error while installing eslint-config-airbnb plugin for ESLint.

During the linter config in my redux based project I am facing the problem related to Eslint config installation.

When I run `npm install --save-dev eslint@7.11.x eslint-config-airbnb@18.1.x eslint-plugin-import@2.22.x eslint-plugin-jsx-a11y@6.2.x eslint-plugin-react@7.20.x eslint-plugin-react-hooks@2.5.x babel-eslint@10.1.x` , I am getting this error:

```
Debugger attached.
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: react-bookstore-cms@0.1.0
npm ERR! Found: eslint@7.11.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"7.11.x" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^5.16.0 || ^6.8.0" from eslint-config-airbnb@18.1.0
npm ERR! node_modules/eslint-config-airbnb
npm ERR!   dev eslint-config-airbnb@"18.1.x" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/dere/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/dere/.npm/_logs/2021-03-22T19_38_55_657Z-debug.log
Waiting for the debugger to disconnect...
```
I have created the project with `create-react-app` script. Even to solve the problem, I removed package-lock.json file and run `npm install` again too. It didn't solve the problem.

--- a/html-css/.github/workflows/linters.yml
+++ b/html-css/.github/workflows/linters.yml
@@ -15,7 +15,7 @@ jobs:
         with:
           node-version: "12.x"
       - name: Setup Lighthouse
-        run: npm install -g @lhci/cli@0.4.x
+        run: npm install -g @lhci/cli@0.7.x
       - name: Lighthouse Report
         run: lhci autorun --upload.target=temporary-public-storage --collect.staticDistDir=.
   webhint:
@@ -28,7 +28,7 @@ jobs:
           node-version: "12.x"
       - name: Setup Webhint
         run: |
-          npm install --save-dev hint@6.0.x
+          npm install --save-dev hint@6.x
           [ -f .hintrc ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/html-css/.hintrc
       - name: Webhint Report
         run: npx hint --telemetry=off .
@@ -42,7 +42,7 @@ jobs:
           node-version: "12.x"
       - name: Setup Stylelint
         run: |
-          npm install --save-dev stylelint@13.3.x stylelint-scss@3.17.x stylelint-config-standard@20.0.x stylelint-csstree-validator
+          npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
           [ -f .stylelintrc.json ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/html-css/.stylelintrc.json
       - name: Stylelint Report
         run: npx stylelint "**/*.{css,scss}"

--- a/html-css/README.md
+++ b/html-css/README.md
@@ -49,7 +49,7 @@ A customizable linting tool that helps you improve your site's accessibility, sp
 
 1. Run
    ```
-   npm install --save-dev hint@6.0.x
+   npm install --save-dev hint@6.x
    ```
    (not sure how to use npm? Read [this](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)).
 2. Copy [.hintrc](.hintrc) to the root directory of your project.
@@ -68,7 +68,7 @@ A mighty, modern linter that helps you avoid errors and enforce conventions in y
 1. Run
 
    ```
-   npm install --save-dev stylelint@13.3.x stylelint-scss@3.17.x stylelint-config-standard@20.0.x stylelint-csstree-validator
+   npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
    ```
 
    (not sure how to use npm? Read [this](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)).

--- a/javascript/.github/workflows/linters.yml
+++ b/javascript/.github/workflows/linters.yml
@@ -16,7 +16,7 @@ jobs:
           node-version: "12.x"
       - name: Setup ESLint
         run: |
-          npm install --save-dev eslint@6.8.x eslint-config-airbnb-base@14.1.x eslint-plugin-import@2.20.x babel-eslint@10.1.x
+          npm install --save-dev eslint@7.x eslint-config-airbnb-base@14.x eslint-plugin-import@2.x babel-eslint@10.x
           [ -f .eslintrc.json ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/javascript/.eslintrc.json
       - name: ESLint Report
         run: npx eslint .
@@ -30,7 +30,7 @@ jobs:
           node-version: "12.x"
       - name: Setup Stylelint
         run: |
-          npm install --save-dev stylelint@13.3.x stylelint-scss@3.17.x stylelint-config-standard@20.0.x stylelint-csstree-validator
+          npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
           [ -f .stylelintrc.json ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/javascript/.stylelintrc.json
       - name: Stylelint Report
         run: npx stylelint "**/*.{css,scss}"

--- a/javascript/README.md
+++ b/javascript/README.md
@@ -36,7 +36,7 @@ node_modules/
 
 ### ESLint
 
-1. Run `npm install --save-dev eslint@6.8.x eslint-config-airbnb-base@14.1.x eslint-plugin-import@2.20.x babel-eslint@10.1.x` (not sure how to use npm? Read [this](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)).
+1. Run `npm install --save-dev eslint@7.x eslint-config-airbnb-base@14.x eslint-plugin-import@2.x babel-eslint@10.x` (not sure how to use npm? Read [this](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)).
 2. Copy [.eslintrc.json](./.eslintrc.json) to the root directory of your project.
 3. **Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students.**
     - If you think that change is necessary - open a [Pull Request in this repository](../README.md#contributing) and let your code reviewer know about it.
@@ -49,7 +49,7 @@ node_modules/
 1. Run
 
    ```
-   npm install --save-dev stylelint@13.3.x stylelint-scss@3.17.x stylelint-config-standard@20.0.x stylelint-csstree-validator
+   npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
    ```
 
    (not sure how to use npm? Read [this](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)).

--- a/react-redux/.github/workflows/linters.yml
+++ b/react-redux/.github/workflows/linters.yml
@@ -16,7 +16,7 @@ jobs:
           node-version: "12.x"
       - name: Setup ESLint
         run: |
-          npm install --save-dev eslint@7.11.x eslint-config-airbnb@18.2.x eslint-plugin-import@2.22.x eslint-plugin-jsx-a11y@6.2.x eslint-plugin-react@7.20.x eslint-plugin-react-hooks@4.1.x babel-eslint@10.1.x
+          npm install --save-dev eslint@7.x eslint-config-airbnb@18.x eslint-plugin-import@2.x eslint-plugin-jsx-a11y@6.x eslint-plugin-react@7.x eslint-plugin-react-hooks@4.x @babel/eslint-parser@7.x
           [ -f .eslintrc.json ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/react-redux/.eslintrc.json
       - name: ESLint Report
         run: npx eslint .
@@ -30,7 +30,7 @@ jobs:
           node-version: "12.x"
       - name: Setup Stylelint
         run: |
-          npm install --save-dev stylelint@13.3.x stylelint-scss@3.17.x stylelint-config-standard@20.0.x stylelint-csstree-validator
+          npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
           [ -f .stylelintrc.json ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/react-redux/.stylelintrc.json
       - name: Stylelint Report
         run: npx stylelint "**/*.{css,scss}"

--- a/react-redux/.github/workflows/linters.yml
+++ b/react-redux/.github/workflows/linters.yml
@@ -16,7 +16,7 @@ jobs:
           node-version: "12.x"
       - name: Setup ESLint
         run: |
-          npm install --save-dev eslint@7.11.x eslint-config-airbnb@18.1.x eslint-plugin-import@2.22.x eslint-plugin-jsx-a11y@6.2.x eslint-plugin-react@7.20.x eslint-plugin-react-hooks@2.5.x babel-eslint@10.1.x
+          npm install --save-dev eslint@7.11.x eslint-config-airbnb@18.2.x eslint-plugin-import@2.22.x eslint-plugin-jsx-a11y@6.2.x eslint-plugin-react@7.20.x eslint-plugin-react-hooks@4.1.x babel-eslint@10.1.x
           [ -f .eslintrc.json ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/react-redux/.eslintrc.json
       - name: ESLint Report
         run: npx eslint .

--- a/react-redux/README.md
+++ b/react-redux/README.md
@@ -29,7 +29,7 @@ Click on the `Details` link to see the full output and the errors that need to b
 
 ### ESLint
 
-1. Run `npm install --save-dev eslint@7.11.x eslint-config-airbnb@18.2.x eslint-plugin-import@2.22.x eslint-plugin-jsx-a11y@6.2.x eslint-plugin-react@7.20.x eslint-plugin-react-hooks@4.1.x babel-eslint@10.1.x` (not sure how to use npm? Read [this](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)).
+1. Run `npm install --save-dev eslint@7.x eslint-config-airbnb@18.x eslint-plugin-import@2.x eslint-plugin-jsx-a11y@6.x eslint-plugin-react@7.x eslint-plugin-react-hooks@4.x @babel/eslint-parser@7.x` (not sure how to use npm? Read [this](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)).
 2. Copy [.eslintrc.json](./.eslintrc.json) to the root directory of your project.
 3. **Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students.**
     - If you think that change is necessary - open a [Pull Request in this repository](../README.md#contributing) and let your code reviewer know about it.
@@ -42,7 +42,7 @@ Click on the `Details` link to see the full output and the errors that need to b
 1. Run
 
    ```
-   npm install --save-dev stylelint@13.3.x stylelint-scss@3.17.x stylelint-config-standard@20.0.x stylelint-csstree-validator
+   npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
    ```
 
    (not sure how to use npm? Read [this](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)).

--- a/react-redux/README.md
+++ b/react-redux/README.md
@@ -29,7 +29,7 @@ Click on the `Details` link to see the full output and the errors that need to b
 
 ### ESLint
 
-1. Run `npm install --save-dev eslint@7.11.x eslint-config-airbnb@18.1.x eslint-plugin-import@2.22.x eslint-plugin-jsx-a11y@6.2.x eslint-plugin-react@7.20.x eslint-plugin-react-hooks@2.5.x babel-eslint@10.1.x` (not sure how to use npm? Read [this](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)).
+1. Run `npm install --save-dev eslint@7.11.x eslint-config-airbnb@18.2.x eslint-plugin-import@2.22.x eslint-plugin-jsx-a11y@6.2.x eslint-plugin-react@7.20.x eslint-plugin-react-hooks@4.1.x babel-eslint@10.1.x` (not sure how to use npm? Read [this](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)).
 2. Copy [.eslintrc.json](./.eslintrc.json) to the root directory of your project.
 3. **Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students.**
     - If you think that change is necessary - open a [Pull Request in this repository](../README.md#contributing) and let your code reviewer know about it.

--- a/ror/.github/workflows/linters.yml
+++ b/ror/.github/workflows/linters.yml
@@ -13,10 +13,10 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-ruby@v1
         with:
-          ruby-version: 2.6.x
+          ruby-version: 3.0.x
       - name: Setup Rubocop
         run: |
-          gem install --no-document rubocop:'~>0.81.0' # https://docs.rubocop.org/en/stable/installation/
+          gem install --no-document rubocop:'>= 1.0, < 2.0' # https://docs.rubocop.org/en/stable/installation/
           [ -f .rubocop.yml ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/ror/.rubocop.yml
       - name: Rubocop Report
         run: rubocop --color
@@ -30,7 +30,7 @@ jobs:
           node-version: "12.x"
       - name: Setup Stylelint
         run: |
-          npm install --save-dev stylelint@13.3.x stylelint-scss@3.17.x stylelint-config-standard@20.0.x stylelint-csstree-validator
+          npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
           [ -f .stylelintrc.json ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/ror/.stylelintrc.json
       - name: Stylelint Report
         run: npx stylelint "**/*.{css,scss}"

--- a/ror/.github/workflows/linters.yml
+++ b/ror/.github/workflows/linters.yml
@@ -16,7 +16,7 @@ jobs:
           ruby-version: 3.0.x
       - name: Setup Rubocop
         run: |
-          gem install --no-document rubocop:'>= 1.0', '< 2.0' # https://docs.rubocop.org/en/stable/installation/
+          gem install --no-document rubocop -v '>= 1.0, < 2.0' # https://docs.rubocop.org/en/stable/installation/
           [ -f .rubocop.yml ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/ror/.rubocop.yml
       - name: Rubocop Report
         run: rubocop --color

--- a/ror/.github/workflows/linters.yml
+++ b/ror/.github/workflows/linters.yml
@@ -16,7 +16,7 @@ jobs:
           ruby-version: 3.0.x
       - name: Setup Rubocop
         run: |
-          gem install --no-document rubocop:'>= 1.0, < 2.0' # https://docs.rubocop.org/en/stable/installation/
+          gem install --no-document rubocop:'>= 1.0', '< 2.0' # https://docs.rubocop.org/en/stable/installation/
           [ -f .rubocop.yml ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/ror/.rubocop.yml
       - name: Rubocop Report
         run: rubocop --color

--- a/ror/.rubocop.yml
+++ b/ror/.rubocop.yml
@@ -1,4 +1,5 @@
 AllCops:
+  NewCops: enable
   Exclude:
     - "db/**/*"
     - "bin/*" 
@@ -24,7 +25,7 @@ Metrics/AbcSize:
 Metrics/ClassLength:
   Max: 150
 Metrics/BlockLength:
-  ExcludedMethods: ['describe']
+  IgnoredMethods: ['describe']
   Max: 30
 
 Style/Documentation:

--- a/ror/README.md
+++ b/ror/README.md
@@ -36,7 +36,7 @@ node_modules/
 
 ### Rubocop
 
-1. Add `gem 'rubocop', '~>0.81.0'` to `Gemfile` (not sure how to use Gemfile? Read [this](https://bundler.io/v1.15/guides/bundler_setup.html)).
+1. Add `gem 'rubocop', '>= 1.0, < 2.0'` to `Gemfile` (not sure how to use Gemfile? Read [this](https://bundler.io/v1.15/guides/bundler_setup.html)).
 2. Run `bundle install`.
 3. Copy [.rubocop.yml](./.rubocop.yml) to the root directory of your project
 4. **Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students.**
@@ -50,7 +50,7 @@ node_modules/
 1. Run
 
    ```
-   npm install --save-dev stylelint@13.3.x stylelint-scss@3.17.x stylelint-config-standard@20.0.x stylelint-csstree-validator
+   npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
    ```
 
    (not sure how to use npm? Read [this](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)).

--- a/ror/README.md
+++ b/ror/README.md
@@ -36,7 +36,7 @@ node_modules/
 
 ### Rubocop
 
-1. Add `gem 'rubocop', '>= 1.0, < 2.0'` to `Gemfile` (not sure how to use Gemfile? Read [this](https://bundler.io/v1.15/guides/bundler_setup.html)).
+1. Add `gem 'rubocop', '>= 1.0', '< 2.0'` to `Gemfile` (not sure how to use Gemfile? Read [this](https://bundler.io/v1.15/guides/bundler_setup.html)).
 2. Run `bundle install`.
 3. Copy [.rubocop.yml](./.rubocop.yml) to the root directory of your project
 4. **Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students.**

--- a/ruby/.github/workflows/linters.yml
+++ b/ruby/.github/workflows/linters.yml
@@ -14,7 +14,7 @@ jobs:
           ruby-version: ">=3.0.x"
       - name: Setup Rubocop
         run: |
-          gem install --no-document rubocop:'>= 1.0, < 2.0' # https://docs.rubocop.org/en/stable/installation/
+          gem install --no-document rubocop:'>= 1.0', '< 2.0' # https://docs.rubocop.org/en/stable/installation/
           [ -f .rubocop.yml ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/ruby/.rubocop.yml
       - name: Rubocop Report
         run: rubocop --color

--- a/ruby/.github/workflows/linters.yml
+++ b/ruby/.github/workflows/linters.yml
@@ -11,10 +11,10 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-ruby@v1
         with:
-          ruby-version: ">=2.6.x"
+          ruby-version: ">=3.0.x"
       - name: Setup Rubocop
         run: |
-          gem install --no-document rubocop:'~>1.9.0' # https://docs.rubocop.org/en/stable/installation/
+          gem install --no-document rubocop:'>= 1.0, < 2.0' # https://docs.rubocop.org/en/stable/installation/
           [ -f .rubocop.yml ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/ruby/.rubocop.yml
       - name: Rubocop Report
         run: rubocop --color

--- a/ruby/.github/workflows/linters.yml
+++ b/ruby/.github/workflows/linters.yml
@@ -14,7 +14,7 @@ jobs:
           ruby-version: ">=3.0.x"
       - name: Setup Rubocop
         run: |
-          gem install --no-document rubocop:'>= 1.0', '< 2.0' # https://docs.rubocop.org/en/stable/installation/
+          gem install --no-document rubocop -v '>= 1.0, < 2.0' # https://docs.rubocop.org/en/stable/installation/
           [ -f .rubocop.yml ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/ruby/.rubocop.yml
       - name: Rubocop Report
         run: rubocop --color

--- a/ruby/.github/workflows/tests.yml
+++ b/ruby/.github/workflows/tests.yml
@@ -14,6 +14,6 @@ jobs:
       - name: Setup RSpec
         run: |
           [ -f Gemfile ] && bundle
-          gem install --no-document rspec:'>=3.0', '< 4.0'
+          gem install --no-document rspec -v '>=3.0, < 4.0'
       - name: RSpec Report
         run: rspec --force-color --format documentation

--- a/ruby/.github/workflows/tests.yml
+++ b/ruby/.github/workflows/tests.yml
@@ -10,10 +10,10 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-ruby@v1
         with:
-          ruby-version: 2.6.x
+          ruby-version: 3.0.x
       - name: Setup RSpec
         run: |
           [ -f Gemfile ] && bundle
-          gem install --no-document rspec:'~>3.0'
+          gem install --no-document rspec:'>=3.0, < 4.0'
       - name: RSpec Report
         run: rspec --force-color --format documentation

--- a/ruby/.github/workflows/tests.yml
+++ b/ruby/.github/workflows/tests.yml
@@ -14,6 +14,6 @@ jobs:
       - name: Setup RSpec
         run: |
           [ -f Gemfile ] && bundle
-          gem install --no-document rspec:'>=3.0, < 4.0'
+          gem install --no-document rspec:'>=3.0', '< 4.0'
       - name: RSpec Report
         run: rspec --force-color --format documentation

--- a/ruby/README.md
+++ b/ruby/README.md
@@ -44,7 +44,7 @@ Click on the `Details` link of the test action to check the results of your test
 
 ### [RuboCop](https://docs.rubocop.org/en/stable/)
 
-1. Add `gem 'rubocop', '~>0.81.0'` to `Gemfile` (not sure how to use Gemfile? Read [this](https://bundler.io/v1.15/guides/bundler_setup.html)).
+1. Add `gem 'rubocop', '>= 1.0, < 2.0'` to `Gemfile` (not sure how to use Gemfile? Read [this](https://bundler.io/v1.15/guides/bundler_setup.html)).
 2. Run `bundle install`.
 3. Copy [.rubocop.yml](./.rubocop.yml) to the root directory of your project
 4. **Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students.**

--- a/ruby/README.md
+++ b/ruby/README.md
@@ -44,7 +44,7 @@ Click on the `Details` link of the test action to check the results of your test
 
 ### [RuboCop](https://docs.rubocop.org/en/stable/)
 
-1. Add `gem 'rubocop', '>= 1.0, < 2.0'` to `Gemfile` (not sure how to use Gemfile? Read [this](https://bundler.io/v1.15/guides/bundler_setup.html)).
+1. Add `gem 'rubocop', '>= 1.0', '< 2.0'` to `Gemfile` (not sure how to use Gemfile? Read [this](https://bundler.io/v1.15/guides/bundler_setup.html)).
 2. Run `bundle install`.
 3. Copy [.rubocop.yml](./.rubocop.yml) to the root directory of your project
 4. **Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students.**

